### ISSUE TITLE
perf(metal): migrate weight buffers to StorageModePrivate

### ISSUE THREAD

**Issue by ohdearquant** [ohdearquant/lattice#179]
## Problem

All Metal buffers in `metal_qwen35.rs` use `StorageModeShared` (91 occurrences, 0 Private). Weight buffers (embed_tokens, layer_weights Q4/Q8, norm weights) are read-only after model load and should use `StorageModePrivate`.

## Why

Apple's documentation states Private buffers can be placed in GPU-optimal memory layouts and compressed formats unavailable to Shared mode. This reduces GPU cache pressure and improves bandwidth utilization. The KV cache f32→f16 migration (#154) will benefit more when the attention kernel's weight reads don't compete for cache lines.

## Scope

- Weights (Q4/Q8 tensors, norm weights, embed_tokens): `StorageModePrivate` + initial blit from staging
- Activation intermediates (hidden, residual, gate, ffn_out, q, k, v, gdn_* scratch): `StorageModePrivate` 
- Keep `StorageModeShared`: logits (CPU readback), KV cache (if CPU append path stays for #154 migration)

## Implementation

1. Create weights with `StorageModePrivate`
2. Use a one-shot `MTLBlitCommandEncoder` to copy from a staging Shared buffer during model load
3. Switch activation buffers to Private (they're only GPU-read/written)
4. Benchmark before/after — expect intercept improvement, possibly bandwidth improvement

## Evidence

```
grep -c "StorageModeShared" metal_qwen35.rs → 91
grep -c "StorageModePrivate" metal_qwen35.rs → 0
```

ADR-064 gate table includes "all production weights/KV/GDN buffers StorageModePrivate" as a target.

Depends: none (can be done independently of #154)
Related: #77 (contention), #172 (dispatch fusion)

**Comment by ohdearquant** [comment]
Still open at `ae6968a5fe`, and the gap has widened rather than narrowed.

Measured across `crates/` at that ref: `StorageModeShared` appears 191 times, 122 of them in `crates/inference/src/forward/metal_qwen35.rs` alone; `StorageModePrivate` appears zero times anywhere. So every Metal buffer in the tree is still CPU-visible, including the weight buffers this issue is about.

The count is stated with a live control rather than as a bare zero: in the same search, `new_buffer` returns 118 and `MTLResourceOptions` returns 123, so the search does reach the allocation sites and the `StorageModePrivate` zero is a real absence rather than a query that matched nothing.
